### DISCUSSION TITLE
Can I give the results of the solver and correspond to the variables defined by yalmip？

### DISCUSSION THREAD

**Discussion by Geriro36** [yalmip/YALMIP#1046] [General discussion]
Hi,

I wonder if I can change the output of the solver. Then change the **_value_** of the variable finally defined by yalmip.
In my understanding, yalmip implements such a process.

1.Define variables ------ 2.input to the solver ------ 3.The solver starts optimizing--------4.solver output results ------ 5.give the results to the corresponding variables

I learned that using **_savesolverinput, savesolveroutput_**  can get the _**sent to**_ and _**returned from**_ the solver。

Can I give a different result(solveroutput). And let it correspond to the variable I defined with yalmip?

**Comment by johanlofberg** [comment]
I have no idea what you are asking.

If you mean if you can override the solution and set a value of a decision variable after having solved a problem, you can always use assign

**Comment by Geriro36** [comment]
Sorry, my English is very poor.

In fact, I first solved it with yalmip. Then I sent the LP file of the model to my friend. He solved it by Benders method and got a group of different answers. But instead of using yalmip, he directly uses Gurobi to solve it. So only the output of the Solver (like **_savesolveroutput_**)

I know that I can set the value of decision variables through **_assign,_** but I don't know the corresponding relationship of decision variables. So it's hard to use **_assign_**.

Is there a better way to deal with this situation？

**Comment by johanlofberg** [comment]
Assuming nothing weird happens along the way so you actually can guarantee that ordering of variables is kept 

```
[Model,internal] = export(...
assign(recover(internal.used_variables),my_buddies_solution)
```

**Reply by Geriro36** [reply]
Thank you for your answer. It helps me a lot
